packages/cdk/apps/core/app.ts

```diff
@@ -13,6 +14,43 @@ const region: string = process.env.CDK_DEFAULT_REGION!;
 const vpcId = getContextOrDefault<Maybe<string>>(app.node, "VPC_ID");
 const bucketName = getContextOrDefault(app.node, `${APP_ENV_NAME}_BUCKET_NAME`, `${APP_NAME}-${account}-${region}`);
 const createNewBucket = getContextOrDefault(app.node, `CREATE_${APP_ENV_NAME}_BUCKET`, "true").toLowerCase() == "true";
+const ecrImageNames: { descriptionName: string; name: string }[] = [
+  { name: "wes", descriptionName: "WES" },
+  { name: "cromwell", descriptionName: "Cromwell" },
+  { name: "nextflow", descriptionName: "Nextflow" },
+  { name: "miniwdl", descriptionName: "MiniWDL" },
+];
+
+const generateEcrImages = (): ParameterProps[] => {
+  const engineParameters: ParameterProps[] = [];
+
+  for (const ecrImage of ecrImageNames) {
+    engineParameters.push(
+      {
+        name: `${ecrImage.name}/ecr-repo/account`,
+        value: getContext(app.node, `ECR_${ecrImage.name.toUpperCase()}_ACCOUNT_ID`),
+        description: `Account ID of ECR that contains the ${ecrImage.descriptionName} docker image`,
+      },
+      {
+        name: `${ecrImage.name}/ecr-repo/region`,
+        value: getContext(app.node, `ECR_${ecrImage.name.toUpperCase()}_REGION`),
+        description: `Region of ECR that contains the ${ecrImage.descriptionName} docker image`,
+      },
+      {
+        name: `${ecrImage.name}/ecr-repo/tag`,
+        value: getContext(app.node, `ECR_${ecrImage.name.toUpperCase()}_TAG`),
+        description: `Docker tag for the ${ecrImage.descriptionName} image`,
+      },
+      {
+        name: `${ecrImage.name}/ecr-repo/repository`,
+        value: getContext(app.node, `ECR_${ecrImage.name.toUpperCase()}_REPOSITORY`),
+        description: `ECR repository for the ${ecrImage.descriptionName} image`,
+      }
+    );
+  }
```

Comment:
IIRC this is where we set the SSM parameters on `account activate` that are then read when we `context deploy` something that needs some of these images?

Do we need SSM at all for this? Perhaps this is a bigger refactor that should be done in a separate task, but I wonder if we could just code these repositories and accounts into the Context stack, overridable by a context parameter there if we so needed. `agc context deploy myContext --cdk-args "-c ECR_MINIWDL_REPOSITORY=agc/miniwdl-mirror-nightly"`

Concretely, this would remove a reason to have to `account activate` when upgrading AGC. To pick up a new version of a WDL or WES image, you just need to redeploy the context that you want to upgrade.

Reply:
The one downside that I can see with that approach is that if you re-deploy a context then you are changing the underlying images that are being used by AGC to run your workflows which might cause issues in the future if a new version of an engine breaks compatibility and would require users to modify how they interact with AGC (which was why I thought that it was implemented this way to begin with).

I'll dig a little deeper to see if we can make your change.

Reply:
The SSM parameters were a relic of an older design, and we've discussed removing them before though we never got around to actually doing it.

Regarding breaking compatibility, we have some protections around destroying a context if there are running workflows. We could extend that to redeploying the context as well